apps/designer/app/canvas/hovered-instance-connector.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect } from "react";
+import debounce from "lodash.debounce";
```

Comment:
we have been using useDebounce from react-use so far because it also handles cleanup correctly, I wonder if you used this because you didn't know that we have something else

Reply:
I used this because we useDebounce complicates code and deeply integrates with react which we don't need as there is no state or callbacks here.

Reply:
Don't you need to clear debounce in effect cleanup? in case component unmounts before debounce calls the callback?

What's your take on using lodash vs lodash.something?

Reply:
I like lodash.something because it allows us to say which lodash functions we want to use in the application and control the dependency, so that we don't start using every lodash function over time

On the other hand it may result in multiple versions of lodash core in the system and the need to maintain the versions of each function

Reply:
see publishHover.cancel(); for clear

Reply:
Maybe we need our own utils package where specific lodash functions can be exported?

Reply:
`lodash.*` are deprecated. I'd prefer to ditch lodash eventually. It adds too much unnecessary code to bundle.

Reply:
oh, why do you add lodash.debounce then instae of importing from lodash directly?

Reply:
we aren't even using it as it seems

Reply:
but i would still say that we need a common utilities belt so maybe having one as a first step and exporting certain lodash functions is a good idea?

Reply:
I will see later if we can use some alternatives